R/ContextEvaluation.R

```diff
@@ -0,0 +1,78 @@
+#' @title Evaluation Context
+#'
+#' @description
+#' A [CallbackEvaluation] accesses and modifies data during [resample()] and [benchmark()] via the `ContextEvaluation`.
+#' See the section on fields for a list of modifiable objects.
+#' See [callback_evaluation()] for a list of stages that access `ContextEvaluation`.
+#'
+#' @export
+ContextEvaluation = R6Class("ContextEvaluation",
+  inherit = Context,
+  public = list(
+
+    #' @field task ([Task])\cr
+    #' The task to be evaluated.
+    #' The task is unchanged during the evaluation.
+    task = NULL,
+
+    #' @field learner ([Learner])\cr
+    #' The learner to be evaluated.
+    #' The learner contains the models after stage `on_evaluation_before_train`.
+    learner = NULL,
+
+    #' @field resampling [Resampling]\cr
+    #' The resampling strategy to be used.
+    #' The resampling is unchanged during the evaluation.
+    resampling = NULL,
+
+    #' @field param_values `list()`\cr
+    #' The parameter values to be used.
+    #' Is usually only set while tuning.
+    param_values = NULL,
+
+    #' @field sets (`list()`)\cr
+    #' The train and test set.
+    #' The sets are available on stage `on_evaluation_before_train``.
```

Comment:
Maybe instead expose the resampling iter, so we can just get `resampling$test_set(iter)`

Reply:
Yes, when I implemented a callback yesterday, I missed it straight away :D